components/scream/src/control/tests/sc_tests.cpp

```diff
@@ -145,12 +145,263 @@ TEST_CASE ("surface_coupling")
+  coupler.register_import("RRTMGP",           7);
+  coupler.register_import("RRTMGP",           8);
+  coupler.register_import("RRTMGP",           9);
+  coupler.register_import("unused",           10);
+  coupler.register_import("unused",           11);
+  coupler.register_import("unused",           12);
+  coupler.register_import("unused",           13);
+  coupler.register_import("unused",           14);
+  coupler.register_import("unused",           15);
+  coupler.register_import("RRTMGP",           16);
+  coupler.register_import("unused",           17);
+  coupler.register_import("RRTMGP",           18);
+  coupler.register_import("unused",           19);
+  coupler.register_import("unused",           20);
+
+  coupler.register_export("T_mid",            0);
+  coupler.register_export("Sa_ptem",          1);
+  coupler.register_export("z_mid",            2);
+  coupler.register_export("Sa_u",             3);
+  coupler.register_export("Sa_v",             4);
+  coupler.register_export("p_mid",            5);
+  coupler.register_export("Sa_dens",          6);
+  coupler.register_export("qv",               7);
+  coupler.register_export("set_zero",         8);
+  coupler.register_export("precip_liq_surf",  9);
+  coupler.register_export("set_zero",         10);
+  coupler.register_export("set_zero",         11);
+  coupler.register_export("set_zero",         12);
+
+  // Complete setup of importer/exporter
+  coupler.registration_ends(import_raw_data, export_raw_data);
+
+  // Perform import
+  coupler.do_import();
+
+  // Perform export
+  coupler.do_export();
+
+  for (int icol=0; icol<ncols; ++icol) {
+
+    // Check imports
+
+    REQUIRE (surf_latent_flux_h(icol) == import_raw_data[0 + icol*num_total_imports]); // 1st scream import
+    REQUIRE (surf_sens_flux_h  (icol) == import_raw_data[1 + icol*num_total_imports]); // 2nd scream import
+    REQUIRE (surf_u_mom_flux_h (icol) == import_raw_data[3 + icol*num_total_imports]); // 3rd scream import (4th total import)
+    REQUIRE (surf_v_mom_flux_h (icol) == import_raw_data[4 + icol*num_total_imports]); // 4th scream import (5th total import)
+
+    // Check exports
+
+    // These exports are direct values from a scream field
+    REQUIRE (export_raw_data[0  + icol*num_exports] == T_mid_h          (icol,    nlevs-1)); // 1st export
+    REQUIRE (export_raw_data[2  + icol*num_exports] == z_mid_h          (icol,    nlevs-1)); // 3rd export
+    REQUIRE (export_raw_data[3  + icol*num_exports] == horiz_winds_h    (icol, 0, nlevs-1)); // 4th export
+    REQUIRE (export_raw_data[4  + icol*num_exports] == horiz_winds_h    (icol, 1, nlevs-1)); // 5th export
+    REQUIRE (export_raw_data[5  + icol*num_exports] == p_mid_h          (icol,    nlevs-1)); // 6th export
+    REQUIRE (export_raw_data[7  + icol*num_exports] == qv_h             (icol,    nlevs-1)); // 8th export
+    REQUIRE (export_raw_data[9  + icol*num_exports] == precip_liq_surf_h(icol            )); // 10th export
```

Comment:
I was wondering if there's any way to check the fields that are computed from scream fields, such as `Sa_ptem`. I guess it would entail writing a lot of code, which would boil dow to re-do what is done inside the SC class, so maybe not much interesting?